chain/messagepool/messagepool.go

```diff
@@ -474,6 +474,15 @@ func (mp *MessagePool) TryForEachPendingMessage(f func(cid.Cid) error) error {
 }
 
 func (mp *MessagePool) resolveToKey(ctx context.Context, addr address.Address) (address.Address, error) {
+	//if addr is not an ID addr, then it is already resolved to a key
+	if addr.Protocol() != address.ID {
+		return addr, nil
+	}
+	return mp.resolveToKeyFromID(ctx, addr)
+}
+
+func (mp *MessagePool) resolveToKeyFromID(ctx context.Context, addr address.Address) (address.Address, error) {
```

Comment:
I don't think we need a new method here, just add the "is not ID" check to `resolveToKey` and leave it as before?

Reply:
I prefer the way I have it. I think it's easier to read! Definitely happy to defer to your suggestion if you have a preference

Reply:
No strong preference!